packages/dappmanager/src/modules/chains/drivers/ethereum2.ts

```diff
@@ -0,0 +1,130 @@
+import fetch from "node-fetch";
+import memoize from "memoizee";
+import { getPrivateNetworkAlias } from "../../../domains";
+import { urlJoin } from "../../../utils/url";
+import { InstalledPackageData } from "../../../types";
+import { ChainDataResult } from "../types";
+import { safeProgress } from "../utils";
+
+const beaconChainServiceName = "beacon-chain";
+const MIN_SLOT_DIFF_SYNC = 60;
+
+// Wait for Promises to resolve. Do not cache rejections
+// Cache for 1 hour, genesis and config should never change
+const fetchNodeSyncingMemo = memoize(fetchNodeSyncingStatus, {
+  promise: true,
+  maxAge: 3e6
+});
+
+/**
+ * Returns a chain data object for an Ethereum 2.0 Prysm beacon chain node
+ * @param apiUrl = "http://beacon-chain.prysm-pyrmont.dappnode:3500/"
+ */
+export async function ethereum2(
+  dnp: InstalledPackageData
+): Promise<ChainDataResult | null> {
+  const beaconChainContainer = dnp.containers.find(
+    container => container.serviceName === beaconChainServiceName
+  );
+  if (!beaconChainContainer) {
+    throw Error(`${beaconChainServiceName} service not found`);
+  }
+  if (!beaconChainContainer.running) {
+    return null; // OK to not be running, just ignore
+  }
+
+  const containerDomain = getPrivateNetworkAlias(beaconChainContainer);
+
+  // http://beacon-chain.prysm-pyrmont.dappnode:3500/
+  const apiUrl = `http://${containerDomain}:3500`;
+
+  try {
+    const nodeSyncing = await fetchNodeSyncingMemo(apiUrl);
+
+    return parseEthereum2State(nodeSyncing);
+  } catch (e) {
+    return {
+      syncing: false,
+      message: `Could not connect to RPC. ${e.message}`,
+      error: true
+    };
+  }
+}
+
+/**
+ * Parses the response from the beacon node to describe if it's currently syncing or not, and if it is, what block it is up to.
+ */
+export function parseEthereum2State(nodeSyncing: NodeSyncing): ChainDataResult {
+  if (!nodeSyncing || !nodeSyncing.data)
+    return {
+      syncing: false,
+      message: "No node syncing data",
+      error: true,
+      progress: 0
+    };
+
+  const { head_slot, sync_distance, is_syncing } = nodeSyncing.data;
+  const highestBlock = parseInt(head_slot) + parseInt(sync_distance);
+
+  if (highestBlock - parseInt(head_slot) < MIN_SLOT_DIFF_SYNC) {
+    return {
+      syncing: false,
+      error: false,
+      message: `Synced #${head_slot}`
+    };
+  } else {
+    return {
+      syncing: is_syncing,
```

Comment:
Should you use `is_syncing` here? You are deciding if it's synced or not with `MIN_SLOT_DIFF_SYNC`, just set to true

Reply:
why say it is syncing whereas it's not?  there can be up to four different states:
- syncing
- synced
- not syncing + error
- syncing but no error

This would be the fourth case